scripts/populate_ingest/populate_ingest_from_nxn.py

```diff
@@ -0,0 +1,178 @@
+        self.__compare_on_title__()
+
+    def filter(self):
+        self.nxn_data.data = [row for row in self.nxn_data.data if
+                         self.nxn_data.get_value(row, 'Organism').lower() in ["human", "human, mouse", "mouse, human"]]
+        self.nxn_data.data = [row for row in self.nxn_data.data if
+                         any([tech.strip() in ["chromium", "drop-seq", "dronc-seq", "smart-seq2", "smarter",
+                                               "smarter (C1)"] for tech in
+                              self.nxn_data.get_value(row, 'Technique').lower().split("&")])]
+        self.nxn_data.data = [row for row in self.nxn_data.data if self.nxn_data.get_value(row, 'Measurement').lower() == 'rna-seq']
+
+    def __convert__(self, nxn_data_row) -> dict:
+        ingest_project = {
+            'content': {
+                "schema_type": "project",
+                "describedBy": self.ingest_schema
+            },
+            'cellCount': self.nxn_data.get_value(nxn_data_row, 'Reported cells total'),
+            'identifyingOrganisms': [organism.strip() for organism in
+                                     self.nxn_data.get_value(nxn_data_row, 'Organism').split(',')],
+            'isInCatalogue': True,
+            'wranglingNotes': f"Auto imported from nxn db {datetime.today().strftime('%Y-%m-%d')}"
+        }
+
+        # setting project title, project description, funders, contributors and publication and publicationsInfo
+        publication_info = self.europe_pmc.query_doi(self.nxn_data.get_value(nxn_data_row, 'DOI'))
+        if publication_info:
+            publications, info = self.publication_converter.convert(publication_info)
+            ingest_project['content'].update(publications)
+            ingest_project['publicationsInfo'] = info
+
+        # setting project short name
+        ingest_project['content'].setdefault('project_core', {}).setdefault('project_short_name', 'tba')
+
+        # setting accessions
+        ingest_project['content'].update(get_accessions(self.nxn_data.get_value(nxn_data_row, 'DOI')))
+        return ingest_project
+
+    def add_projects(self):
+        added_projects = []
+        for nxn_data_row in self.nxn_data.data:
+            project = self.__convert__(nxn_data_row)
+            logging.info(f'Converted nxn data row: {nxn_data_row} to \n'
+                         f'ingest project: {pprint.pformat(project)}' )
+            if self.write:
+                response = self.ingest_api.new_project(project)
+                uuid = response.get('uuid', {}).get('uuid')
+                added_projects.append(uuid)
+                logging.info(f'added to ingest with uuid {uuid}')
+        return added_projects
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Add projects to ingest from Valentine's / NXN database")
+    parser.add_argument("-u", "--url", type=str, help="Base URL for Ingest API", default=INGEST_URL)
```

Comment:
The url should have a default in the code, overridable by an env variable overridable by commandline.
It is easier and more common for tools such as docker, kubernetes, gitlab etc. to pass environmental information as environment variables.

Reply:
code would look like:

``` python
param = "default_value"
if env_var_exists:
   param = get_from_env()
if command_line_vlue_exists:
   param = get_from_command_line()
```